### PR TITLE
Add checksum files to USGS L1, skip datasets with existing metadata

### PR DESCRIPTION
**Pull request by jeremyh**

### CODE REVIEW

**Comment by jeremyh** [comment]
Added a fix+test for the incorrect folder permissions found by @uchchwhash

**Review by uchchwhash — DISMISSED** [review]
Looks good to me, although I am yet to find the code that sets the executable bit for folders :stuck_out_tongue: ... are you just copying it as is?

File: eodatasets/scripts/recompress.py [hunk 1]
```diff
@@ -185,6 +203,88 @@ def _create_tar_with_files(
...
+        compress_args: Dict,
+        verify: PackageChecksum,
+        tmpdir: Path
+):
+    member, open_member = readable_member
+
+    new_member = copy.copy(member)
+    # Copy with a minimum 664 permission, which is used by USGS tars.
+    # (some of our repacked datasets have only user read permission.)
+    new_member.mode = new_member.mode | 0o664
```
Comment (uchchwhash) [inline eodatasets/scripts/recompress.py:218]: Got it! Thanks @jeremyh!

**Comment by jeremyh** [comment]
Added a few trivial tests for failure cases.

**Review by sixy6e — APPROVED** [review]
Looks good.
Well thought through!

**Review by uchchwhash — APPROVED** [review]

**Pull request merged**